### PR TITLE
Configured Packet Capture + Added Viewer Nodes

### PR DESCRIPTION
**Pull request by chelma**
@awick Thanks for all your help configuring/debugging Arkime, and for your quick fixes of the Arkime code when we found issues!

## Description
* Configured the Arkime Capture Nodes to collect traffic mirrored from the user VPC and offload it to S3.  Had to switch them from ECS Fargate to ECS on EC2 due to limitations of Gateway Load Balancers.  
* Added Arkime Viewer Nodes and configured them to read from S3.

## Issues
* https://github.com/arkime/cloud-demo/issues/5
* https://github.com/arkime/cloud-demo/issues/6
* https://github.com/arkime/cloud-demo/issues/7

## Associated Commits
* https://github.com/arkime/arkime/commit/05ffd97b6aff2a900c7fa032d1caee8cd150131b
* https://github.com/arkime/arkime/commit/64926ee1f24b79cfbc6a0703ef9e5a05c17875a5
* https://github.com/arkime/arkime/commit/87469c2fe9830896d2745a9f375639e4ea0c5e71
* https://github.com/arkime/arkime/commit/f6eaf8bada69937d69b26a12b1bd74f8d15ed47a

## Testing
* Updated some unit tests, but still need to add a fair amount to the Python side
* Ran a `create-cluster` and a `add-vpc` CLI command, then visited the Arkime Viewer dashboard to see if everything worked.  I was able to log in, see sessions, and download PCAP.

### CODE REVIEW

**Review by awick — COMMENTED** [review]
Is there anyway for LB names to have the full cluster name, or is that a CDK issue?

File: cdk-lib/viewer-stacks/viewer-nodes-stack.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        const service = new ecs.FargateService(this, 'Service', {
+            cluster,
+            taskDefinition,
+            minHealthyPercent: 0, // TODO: Speeds up test deployments but need to change to something safer
+            enableExecuteCommand: true
+        });        
+
+        const scaling = service.autoScaleTaskCount({ maxCapacity: 10, minCapacity: 4 });
+        scaling.scaleOnCpuUtilization('CpuScaling', {
+            targetUtilizationPercent: 60,
```
[the remark this answers is not in the document]
Comment (chelma) [reply cdk-lib/viewer-stacks/viewer-nodes-stack.ts:94]: Not sure there is a true standard; I chose this as an arbitrary but plausible initial value.

File: cdk-lib/viewer-stacks/viewer-nodes-stack.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            })],
+            healthCheck: {
+                healthyHttpCodes: "200,401",
+                path: '/',
+                unhealthyThresholdCount: 2,
+                healthyThresholdCount: 5,
+                interval: cdk.Duration.seconds(30),
+            },
+        });
+        listener.addAction('/static', {
```
[the remark this answers is not in the document]
Comment (chelma) [reply cdk-lib/viewer-stacks/viewer-nodes-stack.ts:122]: No clue.  I copied this from some of your example code! xD

File: docker-capture-node/arkime_config.ini [hunk 3]
```diff
@@ -1,14 +1,23 @@
 [default]
```
[the remark this answers is not in the document]
Comment (chelma) [reply docker-capture-node/arkime_config.ini:1]: Will do!

**Comment by chelma** [comment]
> Is there anyway for LB names to have the full cluster name, or is that a CDK issue?

If anything, it would be an ELB issue considering this is public DNS.  I'll poke at it a bit and see what leeway ELB provides.  Obviously, we are allowed to provide some part of the final DNS name (a prefix); could be a short character limit or something though.  We'll see!

File: cdk-lib/viewer-stacks/viewer-nodes-stack.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (awick) [reply cdk-lib/viewer-stacks/viewer-nodes-stack.ts:122]: lol, i probably copied from somewhere

[the remark this answers is not in the document]
Comment (chelma) [reply cdk-lib/viewer-stacks/viewer-nodes-stack.ts:122]: Removed

**Comment by chelma** [comment]
Turns out it is easy to have a specified prefix to the ALB domain.

File: docker-capture-node/arkime_config.ini (hunk 3 above)
[the remark this answers is not in the document]
Comment (chelma) [reply docker-capture-node/arkime_config.ini:1]: Removed

**Review by awick — APPROVED** [review]

**Pull request merged**